spanner/spanner_samples.rb

```diff
@@ -0,0 +1,391 @@
+                       index: "AlbumsByAlbumTitle"
+
+  result.rows.each do |row|
+    puts "#{row[:AlbumId]} #{row[:AlbumTitle]}"
+  end
+  # [END read_data_with_index]
+end
+
+def read_data_with_storing_index project_id:, instance_id:, database_id:
+  # [START read_data_with_storing_index]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  result = client.read "Albums", [:AlbumId, :AlbumTitle, :MarketingBudget],
+                       index: "AlbumsByAlbumTitle2"
+
+  result.rows.each do |row|
+    puts "#{row[:AlbumId]} #{row[:AlbumTitle]} #{row[:MarketingBudget]}"
+  end
+  # [END read_data_with_storing_index]
+end
+
+def read_only_transaction project_id:, instance_id:, database_id:
+  # [START read_only_transaction]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  result = client.read "Albums", [:AlbumId, :AlbumTitle, :SingerId],
+                       single_use: { strong: true }
+
+  result.rows.each do |row|
+    puts "#{row[:AlbumId]} #{row[:AlbumTitle]} #{row[:SingerId]}"
+  end
+  # [END read_only_transaction]
+end
+
+def usage
+    puts <<-usage
+Usage: bundle exec ruby spanner_samples.rb [command] [arguments]
+
+Commands:
+  create_database              <instance_id> <database_id> Create Database
+  insert_data                  <instance_id> <database_id> Insert Data
+  query_data                   <instance_id> <database_id> Query Data
+  read_data                    <instance_id> <database_id> Read Data
+  create_index                 <instance_id> <database_id> Create Index
+  create_storing_index         <instance_id> <database_id> Create Storing Index
+  add_column                   <instance_id> <database_id> Add Column
+  update_data                  <instance_id> <database_id> Update Data
+  query_data_with_new_column   <instance_id> <database_id> Query Data with New Column
+  read_write_transaction       <instance_id> <database_id> Read-Write Transaction
+  query_data_with_index        <instance_id> <database_id> Query Data with Index
+  read_data_with_index         <instance_id> <database_id> Read Data with Index
+  read_data_with_storing_index <instance_id> <database_id> Read Data with Storing Index
+  read_only_transaction        <instance_id> <database_id> Read-Only Transaction
+
+Environment variables:
+  GOOGLE_CLOUD_PROJECT must be set to your Google Cloud project ID
+    usage
+end
+
+def run_sample arguments
+  command     = arguments.shift
+  project_id  = ENV["GOOGLE_CLOUD_PROJECT"]
```

Comment:
Nit* (you can ignore): I'd like to have ENV before or after arguments.shift for grouping.